subprojects/highs.wrap

```diff
@@ -0,0 +1,3 @@
+[wrap-git]
+url = https://github.com/scipy/highs.git
+revision = a0df06fb20f2c67c02cf84d8a3b72862c2ab2b27
```

Comment:
Do we still need the wrap file, now that we have the full submodule? It shouldn't be, if I understand this [comment](https://github.com/scipy/scipy/pull/19255#issuecomment-1745669115) correctly.

Reply:
We should provide both, especially in light of not providing the check in `meson.build` for this (it will use the `wrap` file automatically).

Reply:
This works as intended it looks like; it basically checks out the git submodule automatically. The one minor issue is that it leaves behind an extra file:
```
subprojects/highs/.meson-subproject-wrap-hash.txt
```
If that's added to `.gitignore` then I think we're all good here.

Reply:
That's not actually true it seems - when building an sdist after using the auto-checkout-via-wrapfile, I see:
```
WARNING: Submodule 'subprojects/highs' is not checked out and cannot be added to the dist
```

So you still need to run:
```
$ git submodule update --init
Submodule 'subprojects/highs' (https://github.com/scipy/highs) registered for path 'subprojects/highs'
```
That is then instant. But the annoying thing is that `git status` no longer tells you that you've missed a step. You've got the right checkout, but haven't "connected" it as a submodule. Nothing much goes wrong though as far as I can tell, it's just a little weird.